### PR TITLE
Backenddata product listing

### PR DESCRIPTION
**Pull request by Aazam10**
1.Added data to backend.
2. Created context to fetch data on initial render using useEffect
3. Consumed the context in productListing page to show all products from backend.

Note:I have not made the filter component in jsx yet and the filter section css is not applying properly will be working on that in a separate branch.
Csb Link:https://y1dm2p.csb.app/products

### CODE REVIEW

File: src/pages/Products/ProductListing.css [hunk 1]
```diff
@@ -0,0 +1,174 @@
+:root {
+  --light-grey: #c2c2c2;
+  --background-main: f8f8f8;
```
Comment (AnkurChunekar) [inline src/pages/Products/ProductListing.css:3]: Add # to the hex code

**Review by AtulPant2704 — APPROVED** [review]
You have mentioned in the Note that your would work on the Filters late on but just wanted to advise that extract out all the filters as a separate components it would make your work really easy later on. Rest all looks good.

File: src/pages/Products/ProductListing.jsx [hunk 2]
```diff
@@ -1,7 +1,128 @@
...
+                className="slider"
+                min="0"
+                max="5000"
+              />
+            </div>
+          </div>
+          <div className="category-container">
+            <h3 className="category-title">Category</h3>
+            <div className="checkbox">
+              <label for="checkbox-1">
```
Comment (AnkurChunekar) [inline src/pages/Products/ProductListing.jsx:39]: use `htmlFor` instead of for atrribute here

**Comment by AnkurChunekar** [comment]
aazam pr
Scope of Improvement: you can give position sticky to filters container it can look much better.

**Review by AnkurChunekar — APPROVED** [review]
Rest everything looks great, good work!

File: src/pages/Products/ProductListing.css (hunk 1 above)
Comment (Aazam10) [reply src/pages/Products/ProductListing.css:3]: Will do that thanks

File: src/pages/Products/ProductListing.jsx (hunk 2 above)
Comment (Aazam10) [reply src/pages/Products/ProductListing.jsx:39]: I am going to break the sidebar into small components in next pr will do that then.Thanks for the suggestion

**Pull request merged**